src/components/VersionPanel.js

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { View, StyleSheet, ScrollView } from 'react-native';
+import { useTheme, Dialog, Portal, Text, TouchableRipple } from 'react-native-paper';
+import VersionSelector from './VersionSelector';
+
+export default VersionPanel = ({ version, versionList, setVersion }) => {
+  const colors = useTheme().colors;
+  const [visible, setVisible] = React.useState(false);
+
+  const hideDialog = () => setVisible(false);
+
+  return (
+    <View style={[styles.container, { backgroundColor: colors.card }]}>
+      <Text style={styles.header}>Game Version</Text>
+      <TouchableRipple onPress={() => setVisible(true)}>
+        <VersionSelector version={version} />
+      </TouchableRipple>
+
+      <Portal>
+        <Dialog visible={visible} onDismiss={hideDialog}>
+          <Dialog.ScrollArea>
+            <ScrollView contentContainerStyle={{ paddingTop: 8 }}>
+              {versionList.map((version, i) => (
+                <View key={i} style={{ paddingBottom: 8 }}>
+                  <TouchableRipple
+                    onPress={() => {
+                      setVersion(version);
+                      hideDialog();
+                    }}
+                  >
+                    <VersionSelector version={version} />
+                  </TouchableRipple>
+                </View>
+              ))}
+            </ScrollView>
+          </Dialog.ScrollArea>
+        </Dialog>
```

Comment:
In my opinion, it would look a bit better if you add some vertical margins at the far top and far bottom of the dialog; they can have the same value as the horizontal ones. What is more, there should be some title like "Choose game version" - user must be informed what he is doing right now.